contracts/truefi2/LineOfCreditAgency.sol

```diff
@@ -313,11 +318,31 @@ contract LineOfCreditAgency is UpgradeableClaimable, ILineOfCreditAgency {
                 pool,
                 creditOracle.score(borrower),
                 creditOracle.maxBorrowerLimit(borrower),
-                0,
+                collateralVault.stakedAmount(borrower),
                 totalBorrowed(borrower)
             );
     }
 
+    /**
+     * @dev Returns false iff borrower will stay over borrow limit in all pools when `stakedAmount` is staked
+     */
+    function isOverProFormaLimit(address borrower, uint256 stakedAmount) external override view returns (bool) {
+        ITrueFiPool2[] memory pools = poolFactory.getSupportedPools();
+        uint256 _totalBorrowed = totalBorrowed(borrower);
+        uint8 _score = creditOracle.score(borrower);
+        uint256 _maxBorrowerLimit = creditOracle.maxBorrowerLimit(borrower);
+        for (uint256 i = 0; i < pools.length; i++) {
+            if (
+                borrowed[pools[i]][borrower] > 0 &&
+                creditModel.isOverLimit(pools[i], _score, _maxBorrowerLimit, stakedAmount, _totalBorrowed)
+            ) {
+                return true;
```

Comment:
Although this is already merged I want to point out that we access storage variable `creditModel` in a loop.